.github/workflows/ci.yml

```diff
@@ -0,0 +1,102 @@
+name: Exaile CI
+
+on: [push, pull_request]
+
+jobs:
+
+  lint:
+    timeout-minutes: 1
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: pip install black
+      - name: Check format
+        run: make check_format
+
+  test:
+    strategy:
+      matrix:
+        imgtag:
+          - debian10-python3
+          - fedora31-python3
+          - ubuntu18.04-python3
+          - ubuntu19.10-python3
+    timeout-minutes: 4
+    runs-on: ubuntu-latest
+    container: exaile/exaile-testimg:${{ matrix.imgtag }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build and run tests
+        # The container always runs as root regardless of how we set it up in
+        # the Dockerfile. Exaile has tests for file permissions that fail in
+        # that setup, so we de-escalate manually here.
+        run: |
+          useradd -MN exaile
+          chown -R exaile .
+          export HOME=/tmp/home
+          su -m exaile -c "make BUILDDIR=/tmp/build test test_compile check-doc"
+
+  deploy:
+    if: startsWith(github.ref, 'refs/tags/')
+    needs: [lint, test]
+    timeout-minutes: 2
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: sudo apt-get install -y --no-install-recommends python3-gi
+      - id: dist
+        name: Create source dist archive
+        run: |
+          version=${GITHUB_REF#refs/*/}
+          printf '::set-output name=version::%s\n' "${version}"
+          make DIST_VERSION="$version" dist
+          checksum=$(cd dist && sha256sum --tag *)
+          printf '::set-output name=checksum::%s\n' "${checksum}"
+      - name: Create GitHub release
```

Comment:
IIRC it was because it's a bit simpler than the combination of [create-release + upload-release-asset](https://github.com/actions/upload-release-asset#example-workflow---upload-a-release-asset), which has a bunch of nonsense like having to write the filename twice and manually specify the content-type.

Untested:

```
      - id: create_release
        name: Create release
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          tag_name: ${{ github.ref }}
          release_name: Exaile ${{ steps.dist.outputs.version }}
          body: |
            ```
            ${{ steps.dist.outputs.checksum }}
            ```
          draft: true
      - name: Upload release asset
        uses: actions/upload-release-asset@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          upload_url: ${{ steps.create_release.outputs.upload_url }}
          asset_path: dist/exaile-${{ steps.dist.outputs.version }}.tar.gz
          asset_name: exaile-${{ steps.dist.outputs.version }}.tar.gz
          asset_content_type: application/gzip
```

Which one do you prefer? I don't mind either way.

Reply:
Oh, for some reason I had thought that the create-release action had a mechanism for uploading, but I see that it in fact does not. Your original version is definitely simpler then.